### DISCUSSION TITLE
Face Recognition App

### DISCUSSION THREAD

**Discussion by maevesechrist** [nextcloud/all-in-one#616] [Ideas] [answered]
Are there plans to add support for the [Face Recognition App](https://apps.nextcloud.com/apps/facerecognition). The installation instructions include [steps](https://github.com/matiasdelellis/facerecognition/wiki/Docker) for including the necessary dependencies for docker.

**Comment by szaimen** [comment]
Hello, 

unfortunately we cannot provide all needed dependencies for all apps out there because it would make the project unmaintainable.
So I fear this app is out of scope for this project.

However you might be able to add the required dependency to the Nextcloud container by building the containier manually using https://github.com/nextcloud/all-in-one/tree/main/manual-install.

**Comment by ghost** [comment]
Now that the photo app has been revamped in Nextcloud 25, the face recognition app is quite the focus and is a core feature of nextcloud.  especially after the nextcloud conference last weekend it is clear that this app is an important function for nextcloud.

Can that be reconsidered?

**Reply by szaimen** [reply]
The recognize app will indeed get added automatically with NC25.

**Reply by ghost** [reply]
> The recognize app will indeed get added automatically with NC25.

Love IT!

**Reply by ghost** [reply]
The app has not yet been installed and set up for me.

**Reply by szaimen** [reply]
Yeah you are right. I was wrong because I thought it would be bundled. However you can install it from the apps management in your Nextcloud with basically one click: https://apps.nextcloud.com/apps/recognize

**Reply by ghost** [reply]
The dependencies still need to be installed.  So installing the app alone doesn't work as far as I know

**Reply by szaimen** [reply]
The recognize app should work without additional dependencies.

**Reply by ghost** [reply]
Unfortunately, the app still doesn't work for me and many error messages are displayed

**Reply by szaimen** [reply]
Are you talking about the recognize app, or the facerecognition app? These are different ones. The recognize app can get installed, the facerecognition app not.

**Reply by szaimen** [reply]
What errors do you get, can you post some screenshots or log entries?

**Reply by ghost** [reply]
Yes, I'm talking about the recognize app. Does the app work for you?

**Reply by szaimen** [reply]
What errors do you get, can you post some screenshots or log entries?

**Reply by ghost** [reply]
[core] Fehler: RuntimeException: Classifier process timeout at <<closure>>

0. /var/www/html/custom_apps/recognize/lib/Classifiers/Images/ClusteringFaceClassifier.php line 79
   OCA\Recognize\Classifiers\Classifier->classifyFiles("faces", [OCA\Recognize\D ... "], 120)
1. /var/www/html/custom_apps/recognize/lib/BackgroundJobs/ClassifyFacesJob.php line 41
   OCA\Recognize\Classifiers\Images\ClusteringFaceClassifier->classify([OCA\Recognize\D ... "])
2. /var/www/html/custom_apps/recognize/lib/BackgroundJobs/ClassifierJob.php line 70
   OCA\Recognize\BackgroundJobs\ClassifyFacesJob->classify([OCA\Recognize\D ... "])
3. /var/www/html/custom_apps/recognize/lib/BackgroundJobs/ClassifyFacesJob.php line 33
   OCA\Recognize\BackgroundJobs\ClassifierJob->runClassifier("faces", [11,70504])
4. /var/www/html/lib/public/BackgroundJob/Job.php line 78
   OCA\Recognize\BackgroundJobs\ClassifyFacesJob->run([11,70504])
5. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 103
   OCP\BackgroundJob\Job->start(OC\BackgroundJob\JobList {})
6. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 93
   OCP\BackgroundJob\TimedJob->start(OC\BackgroundJob\JobList {})
7. /var/www/html/cron.php line 152
   OCP\BackgroundJob\TimedJob->execute(OC\BackgroundJob\JobList {}, OC\Log {})

at 2022-12-30T18:30:14+00:00

**Reply by ghost** [reply]
![Screenshot_20221230-194948~2.png](https://user-images.githubusercontent.com/90777102/210103307-64804505-0bb9-493e-95d3-d9619b50c4f0.png)

![Screenshot_20221230-195049~2.png](https://user-images.githubusercontent.com/90777102/210103320-3d9c707f-5f1b-43d9-8fe2-1ebbb40ec691.png)

**Reply by szaimen** [reply]
hm... it looks like you should probably open an issue and post the complete error message here: https://github.com/nextcloud/recognize/issues

**Comment by szaimen** [comment, marked as answer]
This will be handled with https://github.com/nextcloud/all-in-one/issues/1162

**Comment by frederikb96** [comment]
Hi, I am able to use the [Recognize App with sub-native speed](https://github.com/nextcloud/recognize#requirements), but I was unable to run the native mode, since TensorFlow got not installed due to the Alpine image and unsupported glibc. My processor supports the AVX instruction set, so it would be very nice if there is a way to enable the native mode.

I tried to [manually install ](https://github.com/nextcloud/recognize/wiki/Manual-install)TensorFlow within the Alpine container by using some Alpine glibc alternatives, but I did not succeed.

**Comment by frederikb96** [comment]
Since this question is already answered, I started a new discussion here #1616
